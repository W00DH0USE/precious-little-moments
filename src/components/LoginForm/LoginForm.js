import React from 'react';
import { Button, Form, Input2 , Required} from "../Utils/Utils";
import Loader from '../Loader/Loader';

function LoginForm(props) {
  return (
      <>
      {props.isLoading ? <Loader {...props} /> : (
      <>
        <Form className="LoginForm" onSubmit={props.handleLogin}>
          <div role="alert">{props.logInError ? <p style={{ color: "red" }}>Incorrect Email or Password</p> : null}</div>
          <div className="user_name">
            <h2 className="Form__title">Precious Little Moment: <span><em>login</em></span></h2>
            <hr />
            <label htmlFor="LoginForm__user_name">
              Email Address <Required />
            </label>
            <Input2
              required
              type="email"
              name="email"
              id="login-email"
              placeholder="Email Address"
            ></Input2>
          </div>

          <div className="password">
            <label htmlFor="LoginForm__password">
              Password <Required />
            </label>
            <Input2
              required
              name="password"
              type="password"
              id="password"
              placeholder="Password"
              autoComplete="off"
            ></Input2>
          </div>
          <Button className="button demo-button" type="submit">Login</Button>
        </Form>
      </>
      )}
      </>
  )
}

export default LoginForm;