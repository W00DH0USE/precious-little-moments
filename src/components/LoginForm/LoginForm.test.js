import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginForm from './LoginForm';


describe(`Login Form Component`, () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<LoginForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});