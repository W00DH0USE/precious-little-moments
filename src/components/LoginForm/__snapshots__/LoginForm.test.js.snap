// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders without crashing 1`] = `
<section>
  <h3>
    Log In
  </h3>
  <form
    className="login-form"
  >
    <div
      className="grid-container"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        required={true}
        type="email"
      />
    </div>
    <div
      className="grid-container"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="btn-login"
      type="submit"
    >
      CONTINUE
    </button>
  </form>
</section>
`;
